File: python/product.py
from typing_extensions import Self


class Product:
    def __init__(self,code,name,price) -> None:
        self._code = code
        self._price = price 
        self._name = name

    @property
    def code(self):
        return self._code 

    @code.setter 
    def code(self,cod):
        if not isinstance(cod,str):
            print('The code is wrong, try again!')
            return
        self._code = cod

    @property
    def price(self):
        return self._price

    @code.setter 
    def price(self,price_in):
        if not isinstance(price_in,str):
            print('The price is wrong, try again!')
            return
        self._price = price_in

File: rust/src/product.rs

#[derive(Clone, Debug)]
pub struct Product {
    pub code: String,
    pub name: String,
    pub price: f64,
}

impl Product {
    pub fn new() -> Product {
        return Product {
            code: String::new(),
            name: String::new(),
            price: 0.0 
        }
    }    

    pub fn get_code(&self) -> String {
        return self.code.clone();
    }

    pub fn get_name(&self) -> String {
        return self.name.clone();
    }

    pub fn get_price(&self) -> f64 {
        return self.price;
    } 
 }
